spec-tests/src/test/java/org/candlepin/spec/auth/ActivationKeyAuthSpecTest.java

```diff
@@ -0,0 +1,150 @@
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertForbidden;
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertUnauthorized;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.ActivationKeyDTO;
+import org.candlepin.dto.api.v1.OwnerDTO;
+import org.candlepin.dto.api.v1.StatusDTO;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.ActivationKeys;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@SpecTest
+class ActivationKeyAuthSpecTest {
+
+    private static OwnerDTO owner;
+    private static ActivationKeyDTO activationKey;
+
+    @BeforeAll
+    static void beforeAll() throws ApiException {
+        ApiClient userClient = ApiClients.admin();
+        owner = userClient.owners().createOwner(Owners.random());
+        ActivationKeyDTO testActivationKey = ActivationKeys.builder().withOwner(owner).build();
+        activationKey = userClient.owners().createActivationKey(owner.getKey(), testActivationKey);
+    }
+
+    @Test
+    @DisplayName("super admin endpoints should reject unauthenticated requests")
+    void superAdminEndpointsShouldRejectActivationKeyAuth() {
+        var client = ApiClients.activationKey(owner.getKey(), activationKey.getName()).owners();
+
+        assertForbidden(() -> client.createOwner(Owners.random()));
+    }
+
+    @Test
+    @DisplayName("verified endpoints should reject unauthenticated requests")
+    void verifiedEndpointsShouldRejectActivationKeyAuth() {
+        var client = ApiClients.activationKey(owner.getKey(), activationKey.getName()).owners();
+
+        assertForbidden(() -> client.getOwner(owner.getKey()));
+    }
+
+    @Test
+    @DisplayName("security hole should reject unauthenticated requests")
+    void securityHoleEndpointsShouldRejectActivationKeyAuth() {
+        var client = ApiClients.activationKey(owner.getKey(), activationKey.getName()).products();
+
+        assertForbidden(() -> client.getProduct("some_uuid"));
+    }
+
+    @Test
+    @DisplayName("no auth security hole should accept unauthenticated requests")
+    void noAuthSecurityHoleEndpointsShouldAcceptNoAuthRequests() throws ApiException {
+        var client = ApiClients.activationKey(owner.getKey(), activationKey.getName()).status();
+
+        StatusDTO status = client.status();
+
+        assertThat(status).isNotNull();
+    }
+
+    @Test
+    @DisplayName("activation key auth requires owner")
+    void shouldRequireOwner() {
+        var client = ApiClients.noAuth().consumers();
```

Comment:
Nack on everything "var." Java is a strongly-typed language and I would like to keep it that way.

Reply:
+1

Reply:
Done. I just wonder what issue are you hitting with var? Java is a strongly-typed language, type inference does not change that and even with extensive refactoring it's still compile-time checked.

Reply:
Static analysis. I understand how var works, it's just an utter pain to read code that makes heavy use of it, and there are a handful of edge cases where it can cause problems during refactors due to its semi-dynamic nature that I'd really rather not deal with.